src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentEventHandlerLoweringPass.cs

```diff
@@ -148,12 +148,13 @@ private IntermediateNode RewriteUsage(IntermediateNode parent, TagHelperProperty
             }
 
             var attributeName = node.AttributeName;
-            if (node.IsDirectiveAttribute && attributeName.StartsWith("@"))
```

Comment:
Why remove the check here? If a user writes a non-directive attribute with an `@` sign all will go to hell no?

Reply:
But a non-directive attribute will never be lowered to `TaghelperDirectiveAttributeIntermediateNode` which the only thing we are operating on.